mirdata/datasets/tonas.py

```diff
@@ -0,0 +1,409 @@
+    all the frames in which a note or a F0 value is comprised.
+
+    Using this dataset:
+    TONAS dataset can be obtained upon request. Please refer to this link: https://zenodo.org/record/1290722 to
+    request access and follow the indications of the .download() method for a proper storing and organization
+    of the TONAS dataset.
+    When TONAS is used for academic research, we would highly appreciate if scientific publications of works
+    partly based on the TONAS dataset quote the following publication:
+
+"""
+import csv
+import os
+from typing import cast, TextIO, Tuple, Optional
+
+import librosa
+import numpy as np
+
+from mirdata import jams_utils
+from mirdata import core
+from mirdata import annotations
+from mirdata import io
+
+
+BIBTEX = """
+@dataset{cofla_computational_analysis_of_flamenco_2013_1290722,
+    author       = {COFLA (COmputational analysis of FLAmenco music) team},
+    title        = {{TONAS: a dataset of flamenco a cappella sung 
+                     melodies with corresponding manual transcriptions}},
+    month        = mar,
+    year         = 2013,
+    publisher    = {Zenodo},
+    version      = {1.0},
+    doi          = {10.5281/zenodo.1290722},
+    url          = {https://doi.org/10.5281/zenodo.1290722}
+}
+@inproceedings{inproceedings,
+    author = {Mora, Joaquin and Gómez, Francisco and Gómez, Emilia
+              and Borrego, Francisco Javier and Díaz-Báñez, José},
+    year = {2010},
+    month = {01},
+    pages = {351-356},
+    title = {Characterization and Similarity in A Cappella Flamenco Cantes.}
+}
+@ARTICLE{6791736,
+    author = {E. {Gómez} and J. {Bonada}},
+    journal = {Computer Music Journal},
+    title = {Towards Computer-Assisted Flamenco Transcription: An Experimental 
+           Comparison of Automatic Transcription Algorithms as Applied to A 
+           Cappella Singing},
+    year = {2013},
+    volume = {37},
+    number = {2},
+    pages = {73-90},
+    doi = {10.1162/COMJ_a_00180}}
+"""
+
+
+REMOTES = None
+
+DOWNLOAD_INFO = """
+        Unfortunately, the TONAS dataset is not available to be shared openly. However,
+        you can request access to the dataset in the following link, providing a brief
+        explanation of what your are going to use the dataset for:
+        ==> https://zenodo.org/record/1290722
+        Then, unzip the dataset and locate it to {}. If you unzip it into a different path,
+        please remember to set the right data_home when initializing the dataset.
+"""
+
+LICENSE_INFO = """
+The TONAS dataset is offered free of charge for internal non-commercial use only. You can not redistribute it nor 
+modify it. Dataset by COFLA team. Copyright © 2012 COFLA project, Universidad de Sevilla. Distribution rights granted 
+to Music Technology Group, Universitat Pompeu Fabra. All Rights Reserved.
+"""
+
+# NoteData extension for Tonas to support energy annotation
+class NoteData(annotations.NoteData):
```

Comment:
♥️

Reply:
Maybe add a small description on the two extended classes just indicating what you are adding

Reply:
Added docstrings for both extended classes!:)

Reply:
Can we open an issue to remind us to remove these two classes once we extend the base classes to include units?